### PR TITLE
Background color fixes

### PR DESCRIPTION
**Pull request by Mcrich23**
Fixes concerns in #97 by now doing practically the same thing, but for backgrounds.

### CODE REVIEW

**Comment by Mcrich23** [comment]
Technically the first pr to actively use the new compiler conditionals instead of retroactively.

**Comment by danielsaidi** [comment]
Hmmm...seems like some things, like the README isn't properly synced? 🤔

**Comment by Mcrich23** [comment]
I can check. I wanted to bring in the compiler stuff and that could have messed with something

**Comment by Mcrich23** [comment]
@danielsaidi on my end, it says merging to an up to date main only updates those 3 files. I can revert that commit if you would like, and bring in the compiler stuff seperately.

File: README.md
Comment (danielsaidi) [inline README.md:1]: For instance, here it looks as if the PR changes to Swift 5.8, but that is already in `main`:

image

**Comment by danielsaidi** [comment]
It looks strange, because it looks as if the PR changes things that are already in `main`.

**Comment by Mcrich23** [comment]
I went ahead and reverted that squash, it should be better now?

**Comment by danielsaidi** [comment]
Yes, a lot better! :) But...this is already in main:

image

**Comment by Mcrich23** [comment]
yes, but in order to make it compile on the branch, I had to add it.

**Comment by Mcrich23** [comment]
I can remove it, but then you will not be able to test it before merge.

**Comment by danielsaidi** [comment]
You should get it if you rebase on main or merge it. It shouldn't be in the diff, since the main branch already has it.

**Comment by Mcrich23** [comment]
I just reverted package.swift, If you look at the files changed, it should be right. If not, I can just close this pr, and make a new one with the branch fixes.

File: Sources/RichTextKit/Colors/RichTextColorPicker.swift [hunk 1]
```diff
@@ -30,18 +30,28 @@ public struct RichTextColorPicker: View {
     public init(
         icon: Image?,
         value: Binding<Color>,
-        quickColors: [Color] = []
+        quickColors: [Color] = [],
+        type: ColorType = .normal
```
Comment (danielsaidi) [inline Sources/RichTextKit/Colors/RichTextColorPicker.swift:34]: This looks great! I actually had something similar previously, but removed it for simplicity. :)

File: Sources/RichTextKit/Colors/RichTextColorPicker.swift [hunk 2]
```diff
@@ -30,18 +30,28 @@ public struct RichTextColorPicker: View {
...
     ) {
         self.icon = icon
         self.value = value
         self.quickColors = quickColors
+        self.type = type
+    }
+    
+    public enum ColorType {
+        case normal, background
```
Comment (danielsaidi) [inline Sources/RichTextKit/Colors/RichTextColorPicker.swift:43]: I'd suggest having three cases:

```swift
public enum ColorType {

    /// Pick a color without color scheme modifications.
    case standard

    /// Pick a foreground color with color scheme modifications.
    case foregroundColor

    /// Pick a background color with color scheme modifications.
    case backgroundColor
}
```

File: Sources/RichTextKit/Colors/RichTextColorPicker.swift [hunk 3]
```diff
@@ -116,7 +126,17 @@ private extension RichTextColorPicker {
 
     func quickPickerButton(for color: Color) -> some View {
         Button {
-            value.wrappedValue = color
+            if type == .background && ((colorScheme == .dark && color == .black) || (colorScheme == .light && color == .white)) {
```
Comment (danielsaidi) [inline Sources/RichTextKit/Colors/RichTextColorPicker.swift:129]: Change `.background` to `.backgroundColor` if you rename the enum case above :)

File: Sources/RichTextKit/Format/RichTextFormatSheet.swift [hunk 4]
```diff
@@ -62,7 +62,8 @@ public struct RichTextFormatSheet: View {
                         RichTextColorPicker(
                             icon: .richTextColorBackground,
                             value: context.binding(for: .background),
-                            quickColors: .quickPickerColors)
+                            quickColors: .quickPickerColors,
+                            type: .background)
```
Comment (danielsaidi) [inline Sources/RichTextKit/Format/RichTextFormatSheet.swift:66]: Change these to `.backgroundColor` and add `.foregroundColor` to the other pickers.

**Review by danielsaidi — COMMENTED** [review]
Looks great! I just have an enum consideration.

**Comment by danielsaidi** [comment]
@Mcrich23 

I have merged the adjustments - it's SO much better. Thank you so much for your help 🙏

**Pull request merged**